python/_tskitmodule.c

```diff
@@ -6575,6 +6575,51 @@ TableCollection_subset(TableCollection *self, PyObject *args)
     return ret;
 }
 
+/* Forward declaration */
+static PyTypeObject TableCollectionType;
+
+static PyObject *
+TableCollection_union(TableCollection *self, PyObject *args, PyObject *kwds)
+{
+    int err;
+    TableCollection *other = NULL;
+    PyObject *ret = NULL;
+    PyObject *node_mapping = NULL;
+    PyArrayObject *nmap_array = NULL;
+    npy_intp *shape;
+    tsk_size_t num_nodes;
+    tsk_flags_t options = 0;
+    int check_shared = true;
+    int add_populations = true;
+    static char *kwlist[] = {"other", "node_mapping", "check_shared_equality", "add_populations", NULL};
+    
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O!O|ii", kwlist, &TableCollectionType, &other, &node_mapping, &check_shared, &add_populations)) {
+        goto out;
+    }
+    nmap_array = (PyArrayObject *) PyArray_FROMANY(node_mapping, NPY_INT32, 1, 1,
+            NPY_ARRAY_IN_ARRAY);
+    if (nmap_array == NULL) {
+        goto out;
+    }
+    shape = PyArray_DIMS(nmap_array);
+    num_nodes = shape[0];
```

Comment:
Here is where we should check if ``num_nodes`` is the right size. If not, raise a ValueError.